Dockerfile

```diff
@@ -0,0 +1,434 @@
+
+FROM runtime AS core
+
+ARG DATAFED_DIR
+ARG DATAFED_INSTALL_PATH
+ARG BUILD_DIR
+ARG LIB_DIR
+
+# The above should also be available at runtime
+ENV DATAFED_INSTALL_PATH="$DATAFED_INSTALL_PATH"
+ENV          DATAFED_DIR="$DATAFED_DIR"
+ENV            BUILD_DIR="$BUILD_DIR"
+
+# copy necessary shared libraries
+# libprotobuf
+COPY --from=dependencies ${LIB_DIR}/libprotobuf.so.3.17.3.0 ${LIB_DIR}/libprotobuf.so.3.17.3.0
+RUN ln -s ${LIB_DIR}/libprotobuf.so.3.17.3.0 ${LIB_DIR}/libprotobuf.so
+
+# libzmq
+COPY --from=dependencies ${LIB_DIR}/libzmq.so.5.2.4 ${LIB_DIR}/libzmq.so.5.2.4
+RUN ln -s ${LIB_DIR}/libzmq.so.5.2.4 ${LIB_DIR}/libzmq.so.5
+RUN ln -s ${LIB_DIR}/libzmq.so.5     ${LIB_DIR}/libzmq.so
+
+# libsodium
+COPY --from=dependencies ${LIB_DIR}/libsodium.so.23.3.0 ${LIB_DIR}/libsodium.so.23.3.0
+RUN ln -s ${LIB_DIR}/libsodium.so.23.3.0 ${LIB_DIR}/libsodium.so.23
+RUN ln -s ${LIB_DIR}/libsodium.so.23     ${LIB_DIR}/libsodium.so
+
+# libboost-program-options
+COPY --from=dependencies /lib/x86_64-linux-gnu/libboost_program_options.so.1.71.0 ${LIB_DIR}/libboost_program_options.so.1.71.0
+RUN ln -s ${LIB_DIR}/libboost_program_options.so.1.71.0 ${LIB_DIR}/libboost_program_options.so
+
+RUN ldconfig
+
+USER datafed
+
+COPY --chown=datafed:datafed ./scripts/generate_datafed.sh     ${DATAFED_DIR}/scripts/generate_datafed.sh
+COPY --chown=datafed:datafed ./scripts/generate_core_config.sh ${DATAFED_DIR}/scripts/generate_core_config.sh
+COPY --chown=datafed:datafed ./scripts/install_core.sh         ${DATAFED_DIR}/scripts/install_core.sh
+COPY --chown=datafed:datafed ./cmake/Version.cmake             ${DATAFED_DIR}/cmake/Version.cmake
+COPY --from=core-build --chown=datafed:datafed ${BUILD_DIR}/core/docker/entrypoint.sh    ${BUILD_DIR}/core/entrypoint.sh
+COPY --from=core-build --chown=datafed:datafed ${DATAFED_INSTALL_PATH}/core/datafed-core ${DATAFED_INSTALL_PATH}/core/datafed-core
+
+# ENTRYPOINT ["/app/entrypoint.sh"]
+# CMD ["/app/datafed-core","--cfg","/app/datafed-core.cfg"]
+
+FROM runtime AS repo
+
+ARG DATAFED_DIR
+ARG DATAFED_INSTALL_PATH
+ARG BUILD_DIR
+ARG LIB_DIR
+
+# The above should also be available at runtime
+ENV DATAFED_INSTALL_PATH="$DATAFED_INSTALL_PATH"
+ENV          DATAFED_DIR="$DATAFED_DIR"
+ENV            BUILD_DIR="$BUILD_DIR"
+
+WORKDIR /datafed
+
+# copy necessary shared libraries
+# libprotobuf
+COPY --from=dependencies ${LIB_DIR}/libprotobuf.so.3.17.3.0 ${LIB_DIR}/libprotobuf.so.3.17.3.0
+RUN ln -s ${LIB_DIR}/libprotobuf.so.3.17.3.0 ${LIB_DIR}/libprotobuf.so
+
+# libzmq
+COPY --from=dependencies ${LIB_DIR}/libzmq.so.5.2.4 ${LIB_DIR}/libzmq.so.5.2.4
+RUN ln -s ${LIB_DIR}/libzmq.so.5.2.4 ${LIB_DIR}/libzmq.so.5
+RUN ln -s ${LIB_DIR}/libzmq.so.5     ${LIB_DIR}/libzmq.so
+
+# libsodium
+COPY --from=dependencies ${LIB_DIR}/libsodium.so.23.3.0 ${LIB_DIR}/libsodium.so.23.3.0
+RUN ln -s ${LIB_DIR}/libsodium.so.23.3.0 ${LIB_DIR}/libsodium.so.23
+RUN ln -s ${LIB_DIR}/libsodium.so.23     ${LIB_DIR}/libsodium.so
```

Comment:
I wonder if there is a way to use the explicit versions mentioned in the scripts/version_dependencies.sh file instead of hard coding. Or at least having a Docker Build Argument that can be overridden when the docker build command is specified. Perhaps in a follow up PR, might be more work then what we want to do now.